src/CsvHelper/TypeConversion/DoubleConverter.cs

```diff
@@ -7,28 +7,37 @@
 
 namespace CsvHelper.TypeConversion
 {
-	/// <summary>
-	/// Converts a <see cref="double"/> to and from a <see cref="string"/>.
-	/// </summary>
-	public class DoubleConverter : DefaultTypeConverter
-	{
-		/// <summary>
-		/// Converts the string to an object.
-		/// </summary>
-		/// <param name="text">The string to convert to an object.</param>
-		/// <param name="row">The <see cref="IReaderRow"/> for the current record.</param>
-		/// <param name="memberMapData">The <see cref="MemberMapData"/> for the member being created.</param>
-		/// <returns>The object created from the string.</returns>
-		public override object ConvertFromString( string text, IReaderRow row, MemberMapData memberMapData )
-		{
-			var numberStyle = memberMapData.TypeConverterOptions.NumberStyle ?? NumberStyles.Float;
+    /// <summary>
+    /// Converts a <see cref="double"/> to and from a <see cref="string"/>.
+    /// </summary>
+    public class DoubleConverter : DefaultTypeConverter
+    {
+        /// <summary>
+        /// Converts the string to an object.
+        /// </summary>
+        /// <param name="text">The string to convert to an object.</param>
+        /// <param name="row">The <see cref="IReaderRow"/> for the current record.</param>
+        /// <param name="memberMapData">The <see cref="MemberMapData"/> for the member being created.</param>
+        /// <returns>The object created from the string.</returns>
+        public override object ConvertFromString(string text, IReaderRow row, MemberMapData memberMapData)
+        {
+            var numberStyle = memberMapData.TypeConverterOptions.NumberStyle ?? NumberStyles.Float;
 
-			if( double.TryParse( text, numberStyle, memberMapData.TypeConverterOptions.CultureInfo, out var d ) )
-			{
-				return d;
-			}
+            if (double.TryParse(text, numberStyle | NumberStyles.AllowExponent, memberMapData.TypeConverterOptions.CultureInfo, out var d))
+            {
+                return d;
+            }
+            if (text.Equals(double.MaxValue.ToString()))
+            {
+                return double.MaxValue;
+            }
+            if (text.Equals(double.MinValue.ToString()))

```

Comment:
you can reduce string allocations by caching results of `.ToString()` and storing it as private static readonly fields
```csharp
private static readonly string MaxValueText = double.MaxValue.ToString();
private static readonly string MinValueText = double.MinValue.ToString();
```
and then
```csharp
if (text.Equals(MaxValueText))
{
	return double.MaxValue;
}
if (text.Equals(MinValueText))
```